SnailWay/SnailWay.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace SnailWay
+{
+    public class SnailWay
+    {
+        public static int[] CalculateWay(int[][] grid)
+        {
+            List<int> snailWay = new List<int>();
+            var tempGrid = grid as IEnumerable<IEnumerable<int>>;
+
+            while (tempGrid.Count() != 0)
+            {
+                snailWay.AddRange(tempGrid.ElementAt(0));
+                tempGrid = tempGrid.Skip(1);
+                snailWay.AddRange((tempGrid.Select(line => line.Last())));
```

Comment:
Try to write algorithm with indices and not enumerating again and again

Reply:
done